hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1242,7 +1242,7 @@ private OmKeyInfo createFakeDirIfShould(String volume, String bucket,
           // in different volume/bucket, such as "/vol1/bucket2/dir2/key2".
           if (key.startsWith(targetKey)) {
             if (!Objects.equals(key, targetKey)
-                && !isKeyDeleted(key, keyTable)) {
+                && keyTable.isExist(key)) {
```

Comment:
@chungen0126 Thanks for working on this patch, however instead of double checking , let deleted key flow outside of this method to caller and outside , it can be filtered. I think the issue is due to the cache is getting cleared after delete operation and isKeyDeleted will not tell as expected. Pls check my patch [here](https://github.com/apache/ozone/pull/5244). We may need not this extra call of "`keyTable.isExist(key)`"

Reply:
@devmadhuu Thanks for your review, here is an issue about creating a deleteSet which will lead to bloat in memory. I think it is a trade-off, I'm not sure which resource is more important.

Reply:
Thanks @chungen0126 @devmadhuu .

Should we consider this PR superceded by #5244 ? Since the latter is fixing the same issue in a cleaner way

Reply:
I am not sure if isExists() API for checking the existence of key till it finds target key will impact performance, as cache is always faster. I applied the patch and ran multiple iterations of each job run in CI for single test class `TestOzoneFileSystem` and I can notice the time taken in completing the test is slightly increased.